### PR TITLE
Refactor: Move middleware to root of server

### LINKED ISSUE #62: Next.js NestedMiddlewareError
https://app.dework.xyz/nation3/app-2?taskId=a3354a32-12f8-4078-9edb-0d3e9adf2bf2

Error when upgrading Next.js from 12.1.6 to 12.2.5:

> Build error occurred
NestedMiddlewareError: Nested Middleware is not allowed, found:
pages/api/_middleware
Please move your code to a single file at /middleware instead.
Read More - https://nextjs.org/docs/messages/nested-middleware

_Originally posted by @aahna-ashina in https://github.com/nation3/mobile-passport/pull/51#pullrequestreview-1083625871_

### PR DESCRIPTION
**Pull request by mikelee**
This change moves nested middleware to the root which prevented upgrading to Next.js v12.2.5. Closes #62 

closes #51

### CODE REVIEW

**Comment by aahna-ashina** [comment]
@xPi2 Could you authorize the Vercel deployment?

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
@mikelee Basic Auth stopped working.

**Comment by mikelee** [comment]
@aahna-ashina I updated next.js to 12.2.5 so the middleware will run. However, the sign in form doesn't pop up and the authorization header is null. I'm not sure what the problem is. Also, next.js doesn't allow a response to be returned from middleware in 12.2. They suggest redirecting to a login form or api route. I'm not what direction you want to go with it.

**Comment by mikelee** [comment]
@aahna-ashina I got basic auth working by rewiting to an api route instead of returning a response.

**Comment by aahna-ashina** [comment]
Thank you for the PR, @mikelee :-D

Looks like the integration tests expect "Unauthorized" to be present in the response body:  https://github.com/nation3/mobile-passport/runs/8167680193?check_suite_focus=true

Would it be possible to add this to `middleware.ts`?

**Comment by mikelee** [comment]
> Thank you for the PR, @mikelee 😄
> 
> Looks like the integration tests expect "Unauthorized" to be present in the response body: https://github.com/nation3/mobile-passport/runs/8167680193?check_suite_focus=true
> 
> Would it be possible to add this to `middleware.ts`?

@aahna-ashina I added the "Unauthorized" response in unauthorized.ts since next.js doesn't allow responses from middleware in v12.2. All integration tests pass now.

**Review by aahna-ashina — APPROVED** [review]

**Comment by aahna-ashina** [comment]
> I added the "Unauthorized" response in unauthorized.ts since next.js doesn't allow responses from middleware in v12.2. All integration tests pass now.

Thank you, @mikelee, looks great 😄

By the way, do you have a Dework username?  I got zero results when searching for https://app.dework.xyz/profile/mikelee

**Comment by mikelee** [comment]
> Thank you, @mikelee, looks great 😄
> 
> By the way, do you have a Dework username? I got zero results when searching for https://app.dework.xyz/profile/mikelee

@aahna-ashina  Thanks. I didn't have an account, but I just made one.
https://app.dework.xyz/profile/mikelee

**Comment by aahna-ashina** [comment]
> Thanks. I didn't have an account, but I just made one.
> https://app.dework.xyz/profile/mikelee

Thank you, @mikelee.  Can you also follow the Nation3 DAO at https://app.dework.xyz/nation3, please?  That way, we can add you to the Dework task.

**Comment by mikelee** [comment]
> Thank you, @mikelee. Can you also follow the Nation3 DAO at https://app.dework.xyz/nation3, please? That way, we can add you to the Dework task.
@aahna-ashina Yup. Done.

**Comment by aahna-ashina** [comment]
> Yup. Done.

@mikelee I don't think it worked.  After you click "Follow", Nation3 should show up on your profile.

**Comment by mikelee** [comment]
> @mikelee I don't think it worked? After you click "Follow," Nation3 should show up on your profile: https://app.dework.xyz/profile/mikelee

That's weird. Nation3 is in my sidebar and I have the option to unfollow.

File: server/middleware.ts [hunk 1]
```diff
@@ -39,12 +40,8 @@ export function middleware(req: NextRequest, event: NextFetchEvent) {
 
     if (!authorizationHeader || wrongCredentials) {
       // Perform Basic Auth
```
Comment (johnmark13) [inline server/middleware.ts:42]: This comment doesn't make any sense :D

**Review by johnmark13 — APPROVED** [review]

File: server/middleware.ts (hunk 1 above)
Comment (aahna-ashina) [inline server/middleware.ts:42]:
```suggestion
      // Redirect
```

Comment (aahna-ashina) [reply server/middleware.ts]: @johnmark13 I agree.  Changed it to "Redirect"

**Pull request merged**